hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,71 @@
+MONGO_ENGINE_COMPLETE = '/etc/foreman-installer/.mongo_engine_upgrade'.freeze
+MONGO_DIR = "/var/lib/mongodb"
+
+# check diskspace
+def disk_space
+  logger.info 'Checking available diskspace in /tmp for upgrade'
+  total_space = `df -k --total --exclude-type=tmpfs`.split("\n")[-1].split[3].to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  /var/lib/mongodb/`.split[0].to_i : 0
+  if total_space < mongo_size
+    logger.error "There is not enough free space #{total_space} the size of MongoDB database is #{mongo_size}, please add additional space to /tmp and try again, exiting."
+    kafo.class.exit(1)
+  else
+    logger.debug "There is #{total_space} free space on disk, which is more than the size of the MongoDB database of #{mongo_size}, continuing with upgrade"
+  end
+end
+
+def migration
+  export_dir = '/tmp/engine_upgrade'
+  logger.info 'Stopping services except Mongo'
+
+  # Verify to stop services before starting migration
+  response = ask('Are you sure you want to continue? This will stop all the services and will take a while depending on the size of your Mongo database.')
```

Comment:
I'm not sure about an interactive prompt. Should we at least have a way to skip it from the command line? I'd also argue the pre validation should do this rather than the actual hook.